### PR TITLE
fix sample parameter when calling binary.fromstring

### PR DESCRIPTION
**Pull request by RodgeFu**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/39437

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
Could you please see if we have this issue in the DPG sample generator? That is the sample generator we will only have after the consolidation work.
BTW Alexander's PR might include this part, but I am not quite sure. You could take a look if you are interested: https://github.com/Azure/autorest.csharp/pull/3726

File: src/AutoRest.CSharp/MgmtTest/Extensions/CodeWriterExtensions.cs [hunk 1]
```diff
@@ -204,10 +204,18 @@ private static CodeWriter AppendDictionaryValue(this CodeWriter writer, CSharpTy
         private static CodeWriter AppendBinaryData(this CodeWriter writer, ExampleValue exampleValue)
         {
             // determine which method on BinaryData we want to use to serialize this BinaryData
-            string method = exampleValue.RawValue != null && exampleValue.RawValue is string ? "FromString" : "FromObjectAsJson";
-            writer.Append($"{typeof(BinaryData)}.{method}(");
-            writer.AppendAnonymousObject(exampleValue);
-            return writer.AppendRaw(")");
+            if (exampleValue.RawValue != null && exampleValue.RawValue is string strValue)
+            {
+                var method = "FromString";
```
[the remark this answers is not in the document]
Comment (RodgeFu) [reply src/AutoRest.CSharp/MgmtTest/Extensions/CodeWriterExtensions.cs:209]: added

**Comment by RodgeFu** [comment]
I think the test part should cover the regression. isn't it?

**Comment by RodgeFu** [comment]
> Could you please see if we have this issue in the DPG sample generator? That is the sample generator we will only have after the consolidation work. BTW Alexander's PR might include this part, but I am not quite sure. You could take a look if you are interested: #3726
I think the test part should cover the regression. isn't it?

**Review by live1206 — APPROVED** [review]

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**